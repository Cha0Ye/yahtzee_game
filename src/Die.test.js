import React from 'react';
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Die from './Die';

// smoke test
it('renders without crashing', () => {
  mount(<Die />);
});

// snapshot test unlocked
it("matches snapshot for unlocked", function() {
  let wrapper = mount(<Die locked={false} val={2}/>);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

// snapshot test locked
it("matches snapshot for locked", function() {
  let wrapper = mount(<Die locked={true} val={2}/>);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});