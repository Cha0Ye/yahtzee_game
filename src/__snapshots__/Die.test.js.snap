// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for locked 1`] = `
<Die
  locked={true}
  val={2}
>
  <button
    className="Die Die-locked"
    onClick={[Function]}
  >
    2
  </button>
</Die>
`;

exports[`matches snapshot for unlocked 1`] = `
<Die
  locked={false}
  val={2}
>
  <button
    className="Die"
    onClick={[Function]}
  >
    2
  </button>
</Die>
`;
